### PR TITLE
bench: add config for circleCI

### PR DESCRIPTION
**Pull request by WenheLI**
This PR adds supports for circleci benchmarking scripts

### CODE REVIEW

**Review by yorkie — CHANGES_REQUESTED** [review]
Shall we run the task when PR updates?

**Comment by yorkie** [comment]
And I think we can't run it on the current main branch: https://app.circleci.com/pipelines/github/alibaba/pipcook

**Comment by WenheLI** [comment]
> https://app.circleci.com/pipelines/github/alibaba/pipcook

Set it only to run CI when PR is submitted.

File: tools/benchmark.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+            --arg esbuild_time $esbuild_time \
+            --arg build_time $build_time \
+            --arg init_time $init_time \
+            --arg test_time $test_time \
+            --arg time_stamp $(date +%s) \
+            --arg mnist_time $mnist_time \
+            '. + [{install_time:$install_time, init_time:$init_time, esbuild_time:$esbuild_time, build_time:$build_time, test_time:$test_time, mnist_time:$mnist_time, timestamp: $time_stamp }]')  > data.json
+
+git config user.email "eric.lwh@alibaba-inc.com"
+git config user.name "WenheLI"
```
Comment (yorkie) [inline tools/benchmark.sh:46]: Could the above lines be configurable?

File: tools/benchmark.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+git add data.json
+git commit --allow-empty  -am"[circleci]: update data"
+git push -q https://${TOKEN}@github.com/imgcook/pipcook-benchmark.git
```
Comment (yorkie) [inline tools/benchmark.sh:49]: I don't expect the result will be pushed to the benchmark repo if I run this locally, how about `sh tools/benchmark.sh --token=xxx`?

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+version: 2.1
+orbs:
+  node: circleci/node@3.0.0
+
+jobs:
+  bench:
```
Comment (yorkie) [inline .circleci/config.yml:6]:
```suggestion
  benchmark:
```

**Comment by yorkie** [comment]
We need documentation about this change.

File: tools/benchmark.sh (hunk 1 above)
Comment (WenheLI) [reply tools/benchmark.sh]: Make it configurable via CircleCI

File: tools/benchmark.sh [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+            --arg build_time $build_time \
+            --arg init_time $init_time \
+            --arg test_time $test_time \
+            --arg time_stamp $(date +%s) \
+            --arg mnist_time $mnist_time \
+            '. + [{install_time:$install_time, init_time:$init_time, esbuild_time:$esbuild_time, build_time:$build_time, test_time:$test_time, mnist_time:$mnist_time, timestamp: $time_stamp }]')  > data.json
+
+if [ -z ${UPLOAD} ]
+then
+  echo "benchmark has been saved to data.json!"
```
Comment (yorkie) [inline tools/benchmark.sh:60]: How about print the output only when not uploading?

File: tools/benchmark.sh [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+else 
+  git config user.email ${EMAIL}
+  git config user.name ${USERNAME}
+  git add data.json
+  git commit --allow-empty  -am"[circleci]: update data"
+  git push -q https://${TOKEN}@github.com/imgcook/pipcook-benchmark.git
+fi
```
Comment (yorkie) [inline tools/benchmark.sh:67]: EOF

File: tools/benchmark.sh [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+test_time=$((t2-t1))
+
+t1=$(date +%s)
+./packages/cli/dist/bin/pipcook run ./example/pipelines/mnist-image-classification.json
+t2=$(date +%s)
+mnist_time=$((t2-t1))
+
+if [ -z ${UPLOAD} ]
+then
+  echo "{\"install_time\":${install_time}, \"init_time\":$init_time, \"esbuild_time\":$esbuild_time, \"build_time\":$build_time, \"test_time\":$test_time, \"mnist_time\":$mnist_time, \"timestamp\": $time_stamp }" | jq
```
Comment (yorkie) [inline tools/benchmark.sh:47]: This line is too long.

File: tools/benchmark.sh [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+  echo $(cat data.json | 
+          jq --arg install_time $install_time \
+            --arg esbuild_time $esbuild_time \
+            --arg build_time $build_time \
+            --arg init_time $init_time \
+            --arg test_time $test_time \
+            --arg time_stamp $(date +%s) \
+            --arg mnist_time $mnist_time \
+            '. + [{install_time:$install_time, init_time:$init_time, esbuild_time:$esbuild_time, build_time:$build_time, test_time:$test_time, mnist_time:$mnist_time, timestamp: $time_stamp }]')  > data.json
```
Comment (yorkie) [inline tools/benchmark.sh:60]: Ditto.

File: tools/benchmark.sh [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+  git config user.email ${EMAIL}
+  git config user.name ${USERNAME}
+  git add data.json
+  git commit --allow-empty  -am\"[circleci]: update data"
```
Comment (yorkie) [inline tools/benchmark.sh:65]:
```suggestion
  git commit --allow-empty -am\"update data"
```

**Review by yorkie — CHANGES_REQUESTED** [review]

File: .circleci/config.yml [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+    docker:
+        - image: cimg/node:14.11.0
+    steps:
+      - checkout
+      - run: bash ./tools/benchmark.sh
+
+workflows:
+  node-bench:
+    jobs:
+      - bench
```
Comment (yorkie) [inline .circleci/config.yml:16]:
```suggestion
      - benchmark
```

**Comment by yorkie** [comment]
@WenheLI What are the hooks to run this script in CircleCI?

**Comment by WenheLI** [comment]
> @WenheLI What are the hooks to run this script in CircleCI?

This will run after every `PR` is merged.

**Comment by yorkie** [comment]
> > @WenheLI What are the hooks to run this script in CircleCI?
> 
> This will run after every `PR` is merged.

It seems incorrect, we should run when PR updates.

**Comment by WenheLI** [comment]
> > > @WenheLI What are the hooks to run this script in CircleCI?
> > 
> > 
> > This will run after every `PR` is merged.
> 
> It seems incorrect, we should run when PR updates.

Humm, that's true. I will update it later

**Comment by WenheLI** [comment]
@yorkie Just came into my mind, since we want to keep recording the benchmark data, we should only run this benchmark when pr is merged. We can run another job for benchmarking on every commit.

**Comment by yorkie** [comment]
Ok, then we should also declare the hooks explicitly that we used.

**Comment by WenheLI** [comment]
> Ok, then we should also declare the hooks explicitly that we used.

Limit the CI only upload benchmark results when committing to master. In other cases, it will only print out the result.

**Comment by yorkie** [comment]
> > Ok, then we should also declare the hooks explicitly that we used.
> 
> Limit the CI only upload benchmark results when committing to master. In other cases, it will only print out the result.

See https://circleci.com/docs/2.0/configuration-cookbook/#branch-filtering-for-job-steps, we need to config it in circle config.

**Review by yorkie — APPROVED** [review]

**Pull request merged**